include/arpa/nameser.h

```diff
@@ -308,15 +309,44 @@ typedef enum __ns_cert_types {
 #define NS_PUT16(s, cp) ns_put16((s), ((cp)+=2)-2)
 #define NS_PUT32(l, cp) ns_put32((l), ((cp)+=4)-4)
 
-unsigned ns_get16(const unsigned char *);
-unsigned long ns_get32(const unsigned char *);
-void ns_put16(unsigned, unsigned char *);
-void ns_put32(unsigned long, unsigned char *);
-
-int ns_initparse(const unsigned char *, int, ns_msg *);
-int ns_parserr(ns_msg *, ns_sect, int, ns_rr *);
-int ns_skiprr(const unsigned char *, const unsigned char *, ns_sect, int);
-int ns_name_uncompress(const unsigned char *, const unsigned char *, const unsigned char *, char *, size_t);
+// ns_get16 returns as an unsigned the concatenation of the 2 bytes pointed by cp in reversed order.
+unsigned ns_get16(const unsigned char *cp : count(2));
+
+// ns_get32 returns as an unsigned long the concatenation of the 4 bytes pointed by cp in reversed order.
+unsigned long ns_get32(const unsigned char *cp : count(4));
+
+// ns_get16 puts l as 2 bytes in reverse to cp.
+void ns_put16(unsigned l, unsigned char *cp : count(2));
+
+// ns_get32 puts l as 4 bytes in reverse to cp.
+void ns_put32(unsigned long l, unsigned char *cp : count(4));
+
+// ns_initparse fills in the data structure pointed to by handle using the response message
+// buffer msg of size len.
+int ns_initparse(const unsigned char *msg : count(len) itype(_Array_ptr<const unsigned char>),
+	int len,
+	ns_msg *handle : itype(_Ptr<ns_msg>));
+
+// ns_parserr extracts information about a response record pointed by handle and stores it
+// in rr, which is a parameter passed to other name server libarary routines.
+int ns_parserr(ns_msg *handle : itype(_Ptr<ns_msg>),
+	ns_sect section,
+	int rrnum,
+	ns_rr *rr : itype(_Ptr<ns_rr>));
+
+int ns_skiprr(const unsigned char *cp : bounds(cp, eom),
```

Comment:
Should ns_skiprr have a comment as well?

Reply:
I'm not clear about what this function does. Unlike the other functions, there is little documentation about `ns_skiprr`  I can find on the internet. The bounds on the parameter `ptr` is not hard to figure out though.

Reply:
I think the network files in MUSL are very poorly documented in general. So it might take more time than expected for me to understand these functions.